frontends/ol-components/src/components/Button/Button.tsx

```diff
@@ -87,9 +92,15 @@ const sizeStyles = (size: ButtonSize, hasBorder: boolean, theme: Theme) => {
   ]
 }
 
-const buttonStyles = (props: ButtonStyleProps & { theme: Theme }) => {
+type ButtonProps = ButtonStyleProps & React.ComponentProps<"button">
+type AnchorProps = ButtonStyleProps & React.ComponentProps<"a">
+type LinkProps = ButtonStyleProps & React.ComponentProps<typeof Link>
+
+const buildStyles = (
+  props: (ButtonProps | AnchorProps | LinkProps) & { theme: Theme },
```

Comment:
Any objection to leaving the types as-is, namely
```tsx
const buildStyles = (props: ButtonStyleProps & { theme: Theme }) => {
```
?

`buildStyles` shouldn't use, e.g., `className` or `onClick`, which both exist on `ButtonProps | AnchorProps | LinkProps`... buildStyles should only use the style-related props.

Reply:
No - I was beefing with Typescript on this branch before the merge from main. They are indeed not needed.